pkg/localstorev2/internal/chunkstore/chunkstore.go

```diff
@@ -53,6 +53,9 @@ type retrievalIndexItem struct {
 	RefCnt    uint8
 }
 
+// TODO original type should be exported
+type RetrievalIndexItemTemp = retrievalIndexItem
```

Comment:
Instead of exporting this type, what we can do is write a package level function here:
```go
func IterateLocations(ctx context.Context, st storage.Store) <-chan Result {
    // do the query here
}
```